ui-lib/src/main/java/co/electriccoin/zcash/ui/BiometricActivity.kt

```diff
@@ -0,0 +1,76 @@
+package co.electriccoin.zcash.ui
+
+import android.content.Context
+import android.content.Intent
+import android.os.Bundle
+import androidx.biometric.BiometricPrompt
+import androidx.core.content.ContextCompat
+import androidx.fragment.app.FragmentActivity
+import co.electriccoin.zcash.ui.common.repository.BiometricRepository
+import co.electriccoin.zcash.ui.common.repository.BiometricResult
+import org.koin.android.ext.android.inject
+
+class BiometricActivity : FragmentActivity() {
+    private val biometricRepository by inject<BiometricRepository>()
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        val requestCode = intent.getStringExtra(EXTRA_REQUEST_CODE).orEmpty()
+        val subtitle = intent.getStringExtra(EXTRA_SUBTITLE).orEmpty()
+
+        val biometricPrompt =
+            BiometricPrompt(
+                this,
+                ContextCompat.getMainExecutor(application),
+                object : BiometricPrompt.AuthenticationCallback() {
+                    override fun onAuthenticationError(
+                        errorCode: Int,
+                        errString: CharSequence
+                    ) {
+                        super.onAuthenticationError(errorCode, errString)
+                        biometricRepository.onBiometricResult(BiometricResult.Failure(requestCode))
+                        finish()
+                    }
+
+                    override fun onAuthenticationSucceeded(result: BiometricPrompt.AuthenticationResult) {
+                        super.onAuthenticationSucceeded(result)
+                        biometricRepository.onBiometricResult(BiometricResult.Success(requestCode))
+                        finish()
+                    }
+
+                    override fun onAuthenticationFailed() {
+                        super.onAuthenticationFailed()
+                        biometricRepository.onBiometricResult(BiometricResult.Failure(requestCode))
+                        finish()
```

Comment:
We should not finish the activity here, as the failed state can be caused, e.g., by a wrong pin set or an unsupported finger used in the biometric authentication.

Reply:
fixed